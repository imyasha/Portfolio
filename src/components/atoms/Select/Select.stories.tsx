import { Select, SelectProps } from './Select';

export default {
	title: 'atoms/Select',
	component: Select,
};

export const SelectStory = (args: SelectProps) => <Select {...args} />;

SelectStory.storyName = 'Select';
SelectStory.args = {
	label: 'Some select',
	id: 'some-select',
	options: ['Option 1', 'Option 2', 'Option 3'],
};
