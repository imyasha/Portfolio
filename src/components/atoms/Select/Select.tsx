import classNames from 'classnames';
import { Field, useField } from 'formik';

export interface SelectProps {
	label: string;
	id: string;
	options: string[];
}

export const Select = ({ label, id, options }: SelectProps) => {
	const fieldProps = { label, id, name: id, options };
	// eslint-disable-next-line @typescript-eslint/no-unused-vars
	const [_, meta] = useField(fieldProps);

	return (
		<label htmlFor={id} className="flex flex-col font-bold mb-2 mt-6">
			{label}
			<Field
				as="select"
				name={id}
				id={id}
				className={classNames(
					'px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700',
					{
						'ring-red-600': meta.touched && meta.error,
					}
				)}
				defaultValue="none"
			>
				<option value="none">Please select</option>
				{options.map((option) => (
					<option value={option} key={option}>
						{option}
					</option>
				))}
			</Field>
			<div className="font-sm font-normal text-red-600">
				{meta.touched && meta.error}
			</div>
		</label>
	);
};
