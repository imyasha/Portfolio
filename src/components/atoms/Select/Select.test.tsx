import { render } from '@testing-library/react';
import { Select } from '../Select';
import { useField } from 'formik';

describe('Select', () => {
	const mockMeta = {
		touched: false,
		error: '',
		initialError: '',
		initialTouched: false,
		initialValue: '',
		value: '',
	};
	const mockField = {
		value: '',
		checked: false,
		onChange: jest.fn(),
		onBlur: jest.fn(),
		multiple: undefined,
		name: 'password',
	};

	beforeEach(() => {
		(useField as jest.Mock).mockReturnValue([mockField, mockMeta]);
	});

	it('renders correctly', () => {
		const { container } = render(
			<Select label="Some select" id="some-select" options={['1', '2', '3']} />
		);
		expect(container).toMatchSnapshot();
	});
});
