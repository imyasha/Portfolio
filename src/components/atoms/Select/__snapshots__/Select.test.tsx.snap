// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div>
  <label
    class="flex flex-col font-bold mb-2 mt-6"
    for="some-select"
  >
    Some select
    <select
      class="px-4 py-2 rounded-md my-2 ring-1 ring-grey-400 dark:ring-grey-700"
      id="some-select"
      name="some-select"
    >
      <option
        selected=""
        value="none"
      >
        Please select
      </option>
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
    </select>
    <div
      class="font-sm font-normal text-red-600"
    />
  </label>
</div>
`;
